addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -589,8 +589,20 @@ var SnippetEditor = Widget.extend({
         // As the 'd-none' class is added to option sections that have no visible
         // options with 'updateOptionsUIVisibility', if no option section is
         // visible, we prevent the activation of the options.
+        // Special case: For now, we only allow activating text options in
+        // translate mode (with no parent editors). These text options have a
+        // special way to be displayed in the editor: We add the options in the
+        // toolbar `onFocus()` and set them back `onBlur()`. Which means the
+        // options section will be empty and get a `d-none` class, while
+        // actually it has visible options (they are just added in the toolbar
+        // DOM). We need to take them into consideration to display options in
+        // translate mode correctly.
         const optionsSectionVisible = editorUIsToUpdate.some(
-             editor => !editor.$optionsSection[0].classList.contains('d-none')
+            (editor) =>
+                !editor.$optionsSection[0].classList.contains("d-none") ||
+                Object.keys(editor.styles).some((key) =>
+                    editor.styles[key].el.closest(".oe-toolbar")
+                )
```

Comment:
Is there no way to make it so that `await Promise.all(editorUIsToUpdate.map(editor => editor.updateOptionsUIVisibility()));` just above resolve to this boolean value? Instead of checking the UI to see if the d-none class was added, just rely on the return value of the related promises?

Reply:
Yes, I changed the code now on `updateOptionsUIVisibility()` to update the options section visibility and resolve with a flag that indicates if the editor has visible options (including the ones displayed in toolbar). I also moved the `website` override of the function to `web_editor`, since it's better to have the **[All options removed from the editor's DOM => Hide options section]** behavior as a global check.